.github/workflows/formatting.yml

```diff
@@ -0,0 +1,49 @@
+name: Formatting Checking
+
+on:
+    push:
+        branches: [ "main", "dev" ]
+    pull_request:
+        branches: [ "main", "dev" ]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    defaults: 
+        run:
+            working-directory: ./evergreen
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python 3.12
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.12
+    - name: Install environment
+    # poetry environment should include:
+    # pip install flake8
+    # pip install autopep8
+    # pip install isort
+    # pip install mypy
+      run: |
+        python -m pip install --upgrade pip
+        pip install poetry
+        poetry install
+
+    - name: Sort imports with isort
+      run: poetry run isort . 
+
+    - name: Format with autopep8
+      run: poetry run autopep8 . 
+
+    - name: Lint with flake8
+      run: poetry run flake8 .
+
+    - name: Find problems from mypy
+      run: poetry run mypy .
```

Comment:
This gets a bunch of `import-untyped` problems from Django. Given that's unavoidable on this project, adding `--disable-error-code import-untyped` will mute those